cacti/tests/test_cacti.py

```diff
@@ -0,0 +1,171 @@
+    'rra[2].xff': 0.5,
+    'ds[mem_buffers].unknown_sec': 0L,
+    'rra[0].pdp_per_row': 1L,
+    'rra[7].rows': 797L,
+    'rra[0].cur_row': 42L,
+    'rra[6].xff': 0.5,
+    'rra[4].pdp_per_row': 1L,
+    'header_size': 2040L,
+    'rra[5].pdp_per_row': 6L,
+    'rra[7].cf': 'MAX',
+    'step': 300L,
+    'rra[3].xff': 0.5,
+    'ds[mem_buffers].max': None,
+    'rra[6].cdp_prep[0].unknown_datapoints': 9L,
+    'rra[4].cdp_prep[0].value': None,
+    'ds[mem_buffers].min': 0.0,
+    'rra[0].cdp_prep[0].value': None,
+    'rra[1].cdp_prep[0].value': 41644.66666666667,
+    'rra[1].cur_row': 234L,
+    'rra[0].rows': 600L,
+    'rra[6].cdp_prep[0].value': 21262.533333333333,
+    'rra[5].cur_row': 194L,
+    'rra[4].rows': 600L,
+    'rra[3].cf': 'AVERAGE',
+    'rra[0].xff': 0.5,
+    'rra[5].cf': 'MAX',
+    'rra[2].cdp_prep[0].unknown_datapoints': 9L,
+    'rra[5].xff': 0.5,
+    'rra[2].cur_row': 742L,
+    'rra[6].cf': 'MAX',
+    'rra[7].cdp_prep[0].unknown_datapoints': 177L,
+    'rra[1].cdp_prep[0].unknown_datapoints': 0L,
+    'rra[2].rows': 775L,
+    'rra[2].pdp_per_row': 24L,
+}
+
+ts = int(time.time())
+MOCK_FETCH = ((ts - 300, ts, 300), ('mem_buffers',), [(2048,), (None,)])
+
+MOCK_RRD_META = [
+    ('localhost', None, '/var/lib/cacti/rra/localhost_mem_buffers_3.rrd'),
+    ('localhost', None, '/var/lib/cacti/rra/localhost_mem_swap_4.rrd'),
+    ('localhost', None, '/var/lib/cacti/rra/localhost_load_1min_5.rrd'),
+    ('localhost', None, '/var/lib/cacti/rra/localhost_users_6.rrd'),
+    ('localhost', None, '/var/lib/cacti/rra/localhost_proc_7.rrd'),
+]
+
+
+CHECK_NAME = 'cacti'
+
+CACTI_CONFIG = {
+    'mysql_host': 'nohost',
+    'mysql_user': 'mocked',
+    'rrd_path': '/rrdtool/is/mocked',
+    'tags': ['optional:tag1'],
+}
+
+
+@pytest.fixture
+def aggregator():
+    from datadog_checks.stubs import aggregator
+    aggregator.reset()
+    return aggregator
+
+
+@pytest.fixture
+def check():
+    return Cacti(CHECK_NAME, {}, {})
+
+
+def testCheck(aggregator, check):
+
+    mocks = [
+        mock.patch('datadog_checks.cacti.cacti.pymysql'),
+        mock.patch('datadog_checks.cacti.cacti.rrdtool'),
+        mock.patch('datadog_checks.cacti.Cacti._get_rrd_info', return_value=MOCK_INFO),
+        mock.patch('datadog_checks.cacti.Cacti._get_rrd_fetch', return_value=MOCK_FETCH),
+        mock.patch('datadog_checks.cacti.Cacti._fetch_rrd_meta', return_value=MOCK_RRD_META),
+    ]
+
+    for mock_func in mocks:
+        mock_func.start()
+
+    # Run the check twice to set the timestamps and capture metrics on the second run
+    check.check(CACTI_CONFIG)
+    check.check(CACTI_CONFIG)
+    print aggregator._metrics
+
+    for mock_func in mocks:
+        mock_func.stop()
+
+    # We are mocking the MySQL call so we won't have cacti.rrd.count or cacti.hosts.count metrics,
+    # check for metrics that are returned from our mock data.
+    aggregator.assert_metric('cacti.metrics.count', value=10, tags=['optional:tag1'])
```

Comment:
Can we make the tags a constant?
Something like:
```
CUSTOM_TAGS = ['optional:tag1']

CACTI_CONFIG = {
    'mysql_host': 'nohost',
    'mysql_user': 'mocked',
    'rrd_path': '/rrdtool/is/mocked',
    'tags': list(CUSTOM_TAGS),
}

# ...

assert_metrics(..., tags=CUSTOM_TAGS)
```

Reply:
Ah, I did ``list(CUSTOM_TAGS)`` because there are some checks that directly manipulate the tags (like ``.append(foobar)``) which can mess up assertions and other stuff, so it's _ok_ to pass in a copy of the tags.